pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -45,23 +50,36 @@ func (r *Reconciler) Reconcile(ctx context.Context, logger logr.Logger, client c
 	if err != nil {
 		return controller.Result{}, err
 	}
+	// desired state
+	// read current state (find config map with cilium config)
+	// think about splitting out upgrade into upgradeVersion and upgradeConfig
+	// extend UpgradePlan to detect configuration changes as well
 
 	if !installation.Installed() {
 		return r.install(ctx, logger, client, spec)
 	}
 
 	logger.Info("Cilium is already installed, checking if it needs upgrade")
 	upgradeInfo := cilium.BuildUpgradePlan(installation, spec)
+
+	logger.Info("Attempting to retrieve ConfigMap")
 	if !upgradeInfo.Needed() {
 		logger.Info("Cilium is already in desired version")
-	} else {
+	} else if upgradeInfo.VersionUpgradeNeeded() {
 		logger.Info("Cilium upgrade needed", "reason", upgradeInfo.Reason())
 
 		if result, err := r.upgrade(ctx, logger, client, installation, spec); err != nil {
 			return controller.Result{}, err
 		} else if result.Return() {
 			return result, nil
 		}
+	} else if upgradeInfo.ConfigUpgradeNeeded() {
+		logger.Info("Cilium config upgrade needed", "reason", upgradeInfo.Reason())
+		if result, err := r.upgradeConfig(ctx, logger, client, installation, spec); err != nil {
+			return controller.Result{}, err
+		} else if result.Return() {
+			return result, nil
+		}
```

Comment:
I don't like how crowded this code reads. Will try to find a way to make it easier to read.